packages/@aws-cdk/aws-iam/lib/policy-document.ts

```diff
@@ -143,13 +143,59 @@ export class AccountRootPrincipal extends AccountPrincipal {
 
 /**
  * A principal representing all identities in all accounts
+ * @deprecated use `AnyPrincipal`
  */
 export class Anyone extends ArnPrincipal {
   constructor() {
     super('*');
   }
 }
 
+/**
+ * A principal representing all identities in all accounts
+ */
+export class AnyPrincipal extends Anyone {
```

Comment:
I would have reversed the `extends` direction -- having the replacement built on top of the deprecated is... questionable.